WooCommerce/Classes/ViewRelated/Products/Edit Product/Cells/ImageAndTitleAndTextTableViewCell.swift

```diff
@@ -21,6 +21,19 @@ final class ImageAndTitleAndTextTableViewCell: UITableViewCell {
         }
     }
 
+    /// View model with a switch in the cell's accessory view.
+    struct SwitchableViewModel {
+        let viewModel: ViewModel
+        let isSwitchOn: Bool
+        let onSwitchChange: (_ isOn: Bool) -> Void
+
+        init(viewModel: ViewModel, isSwitchOn: Bool, onSwitchChange: @escaping (_ isOn: Bool) -> Void) {
+            self.viewModel = viewModel
+            self.isSwitchOn = isSwitchOn
+            self.onSwitchChange = onSwitchChange
+        }
+    }
```

Comment:
I think that for this use case, it could have been used the `SwitchTableViewCell`, the cell `ImageAndTitleAndTextTableViewCell` seems to have too many roles :D

Reply:
yea it might be making `ImageAndTitleAndTextTableViewCell` a bit too versatile, I just thought the `accessoryView` is separate from the cell's content view and the content view is exactly the same. `SwitchTableViewCell` doesn't have the layout that we'd like for the product form 😞 if it gets more complicated, we can move the switch cell to a new cell in the future